app/models/service_order.rb

```diff
@@ -5,14 +5,15 @@ class ServiceOrder < ActiveRecord::Base
 
   belongs_to :tenant
   belongs_to :user
-  has_many :miq_requests, :dependent => :nullify
+  has_many :miq_requests
```

Comment:
I tried this locally and I did not see that issue.  The `before_destroy` callback will run before the nullify happens.

The current test failure issue I am seeing is that the `order_immediately` method sets the state to `ordered` when the instance is created which is causing the `checkout` method to raise an error.